spec/spec_helper.rb

```diff
@@ -152,6 +136,40 @@ def capture_io
   $stdout = STDOUT
 end
 
+# Get a list of running threads, ignoring jruby system threads if necessary.
+def running_threads
+  threads = []
+  Thread.list.each do |t|
+    if RUBY_ENGINE == 'jruby'
+      thread_name = JRuby.reference(t).native_thread.get_name
+      threads << t.inspect unless thread_name == "Finalizer" or thread_name.include?("JRubyWorker")
+    else
+      threads << t.inspect
+    end
+  end
+  return threads
+end
```

Comment:
This is a great enhancement to prevent all the inline checks which you factored out. :+1: